src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -300,4 +300,35 @@ export default class ReimbursementRequestService {
 
     return expense;
   }
+
+  /**
+   * Service function to mark a reimbursement request as delivered
+   * @param submitter is the User marking the request as delivered
+   * @param requestId is the ID of the reimbursement request to be marked as delivered
+   * @throws NotFoundException if the id is invalid or not there
+   * @throws AccessDeniedException if the creator of the request is not the submitter
+   * @returns the updated reimbursement request
+   */
+  static async markReimbursementRequestAsDelivered(submitter: User, reimbursementRequestId: string) {
+    const reimbursementRequest = await prisma.reimbursement_Request.findUnique({
+      where: { reimbursementRequestId }
+    });
+
+    if (!reimbursementRequest) throw new NotFoundException('Reimbursement Request', reimbursementRequestId);
+
+    if (!(reimbursementRequest.dateDelivered === null))
```

Comment:
you can just do if (reimbursementRequest.dateDelivered) here, it automatically converts it to a boolean that checks this